### PR TITLE
[Term Entry] D3 Scaling .scaleBand()

### LINKED ISSUE #3883: [Term Entry] D3 Scaling .scaleBand()
### Reason/inspiration (optional)

We would like a new term entry on the `.scaleBand()` method for [D3](https://www.codecademy.com/resources/docs/d3): Scaling. The entry should go in a new file under `docs/content/d3/concepts/scaling/terms/scaleBand/scaleBand.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```js `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by ShatilKhan**
### Description

Closes #3883 

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/d3/concepts/scaling/terms/scaleBand/scaleBand.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## Syntax
+
+```javascript
+d3.scaleBand()
+  .domain(inputDomain)
+  .range(outputRange)
+  .padding(innerPadding)
+  .paddingOuter(outerPadding);
+```
```
Comment (avdhoottt) [inline content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:25]:
```suggestion
```pseudo
d3.scaleBand()
  .domain(inputDomain)
  .range(outputRange)
  .padding(innerPadding)
  .paddingOuter(outerPadding);
```
```
The syntax is written in pseudo block.

File: content/d3/concepts/scaling/terms/scaleBand/scaleBand.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  .scaleBand()
+  .domain(emojis)
+  .range([0, 200]) // Width of the visualization
+  .padding(0.2); // Padding between emojis
+
+// Log the positions of emojis
+emojis.forEach(function (emoji) {
+  console.log(`Position of ${emoji}: ${xScale(emoji)}`);
+});
+```
```
Comment (avdhoottt) [inline content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:51]: Please add a note that describes the prerequisites to run this code. To my knowledge, the D3 needs to be imported, include that line in the code too.

File: content/d3/concepts/scaling/terms/scaleBand/scaleBand.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+The result of the above code is as follows:
+
+```shell
+    Position of 🍎: 0
+    Position of 🍌: 40
+    Position of 🍇: 80
+    Position of 🍊: 120
+    Position of 🍓: 160
+```
```
Comment (avdhoottt) [inline content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:61]: I'm getting a different output than this, please look into this too.

File: content/d3/concepts/scaling/terms/scaleBand/scaleBand.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+- `inputDomain`: An array representing the input domain (categories or discrete values).
+- `outputRange`: An array representing the output range (typically the width of the visualization).
+- `innerPadding`: Optional. Specifies the padding between bands, as a fraction of the band width (default is 0.).
+- `outerPadding`: Optional. Specifies the padding before the first band and after the last band, as a fraction of the band width (default is 0.).
+
+## Example
+
+The example code below defines an array of emojis, creates a band scale, and logs the x-axis positions of each emoji.
+
```
Comment (avdhoottt) [inline content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:35]:
```suggestion
- `inputDomain`: An array representing the input domain (categories or discrete values).
- `outputRange`: An array representing the output range (typically the width of the visualization).
- `innerPadding`: Optional. Specifies the padding between bands, as a fraction of the bandwidth (default is 0.).
- `outerPadding`: Optional. Specifies the padding before the first band and after the last band, as a fraction of the bandwidth (default is 0.).

## Example

The example code below defines an array of emojis, creates a band scale, and logs the x-axis positions of each emoji.

```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@ShatilKhan I've left some comments for you, please have a look at them. Nice entry👍

**Comment by ShatilKhan** [comment]
yup

**Comment by avdhoottt** [comment]
Please try to do it as soon as you can, it's been a week since I suggested the changes.

File: content/d3/concepts/scaling/terms/scaleBand/scaleBand.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  .domain(inputDomain)
+  .range(outputRange)
+  .padding(innerPadding)
+  .paddingOuter(outerPadding);
+```
+
+- `inputDomain`: An array representing the input domain (categories or discrete values).
+- `outputRange`: An array representing the output range (typically the width of the visualization).
+- `innerPadding`: Optional. Specifies the padding between bands, as a fraction of the band width (default is 0.).
+- `outerPadding`: Optional. Specifies the padding before the first band and after the last band, as a fraction of the band width (default is 0.).
```
Comment (avdhoottt) [inline content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:29]:
```suggestion
- `inputDomain`: An array representing the input domain (categories or discrete values).
- `outputRange`: An array representing the output range (typically the width of the visualization).
- `innerPadding`: Specifies the padding between bands, as a fraction of the bandwidth (default is 0.).
- `outerPadding`: Specifies the padding before the first band and after the last band, as a fraction of the bandwidth (default is 0.).
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@ShatilKhan Hey, Are you going to make the suggested changes?

**Comment by ShatilKhan** [comment]
Done

**Comment by ShatilKhan** [comment]
I'm making the other changes

File: content/d3/concepts/scaling/terms/scaleBand/scaleBand.md (hunk 2 above)
Comment (ShatilKhan) [reply content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:50]: @avdhoottt as you said in my other PR, we don't have to add html
I believe there will be a different pre-requisite section where the setup is shown right?

Comment (avdhoottt) [reply content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:50]: Yes @ShatilKhan, don't add the HTML part as it is not necessary (In the other PR too).

**Comment by avdhoottt** [comment]
@ShatilKhan Please, make all the changes. I can send it to the second review then.

**Comment by ShatilKhan** [comment]
Working on it

File: content/d3/concepts/scaling/terms/scaleBand/scaleBand.md (hunk 3 above)
Comment (ShatilKhan) [reply content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:59]: I ran the code 
image
After saving file as html we can open it in a browser & inspect the elements 
In the inspection tap you'll see the output
image
As this requires some html setup, only using js wouldn't print the right results
So I'll just add a line to explain it

**Review by ShatilKhan — COMMENTED** [review]
Updated Example instructions

File: content/d3/concepts/scaling/terms/scaleBand/scaleBand.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+Finally if you open you're html file in browser & go to console in inspect tool you should see the below output.
+Console Output:
+
+```shell
+    Position of 🍎: 0
+    Position of 🍌: 40
+    Position of 🍇: 80
+    Position of 🍊: 120
+    Position of 🍓: 160
```
[the remark this answers is not in the document]
Comment (ShatilKhan) [reply content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:61]: Yeah I think it's giving the actual position with respect to the pc display size probably 
How about I just add this in the doc?
The actual output 
I also get the same output

[the remark this answers is not in the document]
Comment (ShatilKhan) [reply content/d3/concepts/scaling/terms/scaleBand/scaleBand.md:61]: fixed it, it was due to padding issue

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Looks good for the second review.

**Review by dakshdeepHERE — APPROVED** [review]
Did some minor changes from my side. Now, this looks good to merge.

**Review by avdhoottt — APPROVED** [review]

**Pull request merged**